### PR TITLE
chore: add apple silicon support for buf tooling

### PR DESCRIPTION
**Pull request by backevik**

### CODE REVIEW

File: tools/buf/rules.mk [hunk 1]
```diff
@@ -3,7 +3,14 @@ buf_version := 0.41.0
 buf := $(buf_cwd)/$(buf_version)/bin/buf
 export PATH := $(dir $(buf)):$(PATH)
 
-buf_bin_url := https://github.com/bufbuild/buf/releases/download/v$(buf_version)/buf-$(shell uname -s)-$(shell uname -m)
+buf_arch = $(shell uname -s)-$(shell uname -m)
```
Comment (odsod) [inline tools/buf/rules.mk:6]: imo remove the blank lines since these lines all belong together

**Review by odsod — APPROVED** [review]
LGTM - please update whitespace before merge :+1:

File: tools/buf/rules.mk (hunk 1 above)
Comment (backevik) [reply tools/buf/rules.mk:6]: Updated! Think I understood which whitespace you meant. Please verify!

Comment (odsod) [reply tools/buf/rules.mk:6]: Perfect!

**Pull request merged**